### PR TITLE
Use XZ Compression

### PR DESCRIPTION
**Pull request by doawoo**
## Testing

I confirm that the changes in this PR have been tested on:

- [x] Windows (10 or above)
- [x] MacOS (11.0 or above)
- [x] Linux (glibc)
- [x] Linux (musl)

## Changelog

* Replaces gzip with xz+lzma2 compression

* Reduces the size of the FOILZ archive by moving `mode` to use `c_uint` rather than `u64`

* Prints a warning for missing `7z` in the PATH as we really only use them for Windows builds 

* Fixes up the README including a mistyped build qualifier for custom ERTS slipstream archives  

------

# Supporting Screenshots

Comparison of GZIP vs XZ/LZMA on `MIX_ENV=prod`

```
-rwxr--r--   1 digit  staff   6.2M Mar 23 09:11 example_cli_app_macos_m1_GZIP
-rwxr--r--   1 digit  staff   5.7M Mar 23 08:45 example_cli_app_macos_m1_XZ
```

### CODE REVIEW

**Comment by bsedat** [comment]
To track where the XZ implementation came from is it from https://git.tukaani.org/?p=xz-embedded.git? Would you be okay adding the commit hash of the XZ project that this pulls from to your associated burrito git commit message so we can track future patches more easily?

**Comment by doawoo** [comment]
Good idea, I actually fetched them from a tarball on this page:
https://tukaani.org/xz/embedded.html

I can put a small VERSION file in the xz directory that indicates the version (in this case it would be `20210201` and possibly a URL to the tar + signature of the tar.

Not sure how the git maps 1 to 1 with the tarballs, seeing as there are no "releases".

**Comment by doawoo** [comment]
I think I'll just update it to the tip of tree on their master and log that commit in a `VERSION` file

File: lib/steps/build/pack_and_build.ex [hunk 1]
```diff
@@ -37,7 +37,7 @@ defmodule Burrito.Steps.Build.PackAndBuild do
           {"__BURRITO_IS_PROD", is_prod?()},
           {"__BURRITO_RELEASE_PATH", context.work_dir},
           {"__BURRITO_RELEASE_NAME", release_name},
-          {"__BURRITO_PLUGIN_PATH", plugin_path}
```
Comment (bsedat) [inline lib/steps/build/pack_and_build.ex:40]: Super minor nitpick: this will be changed back by the Elixir formatter.

**Review by bsedat — APPROVED** [review]
I haven't tried it end-to-end yet but this looks good to me.

**Comment by doawoo** [comment]
I'll give this a formatter pass then merge! Thanks

**Pull request merged**